src/backend/core/api/viewsets.py

```diff
@@ -11,6 +10,7 @@
     response,
     viewsets,
 )
+from rest_framework.throttling import UserRateThrottle
```

Comment:
I think we should write a test to asset that it is working, plus to be secure in the future that it is still working when we upgrade something, I wrote a test for you, the test is working, but you can probably make it cleaner:

`./src/backend/core/tests/test_throttle.py`
```py
"""
Test Throttle in People's app.
"""

from .utils import OIDCToken

from django.test.utils import override_settings

import pytest
from rest_framework.test import APIClient

from core import factories, models
from core.api import serializers

from .utils import OIDCToken

pytestmark = pytest.mark.django_db


@override_settings(REST_FRAMEWORK={
  'DEFAULT_THROTTLE_RATES': {
    'burst': '2/min',
    'sustained': '2/min'
}})
def test_throttle():
    identity = factories.IdentityFactory()
    user = identity.user
    contact = factories.ContactFactory(owner=user)
    user.profile_contact = contact
    user.save()
    jwt_token = OIDCToken.for_user(user)

    # Let's have 5 contacts in database:
    assert user.profile_contact is not None  # Excluded because profile contact
    base_contact = factories.BaseContactFactory()  # Excluded because overriden
    factories.ContactFactory(
        base=base_contact
    )  # Excluded because belongs to other user

    client = APIClient()
    # some end point you want to test (in this case it's a public enpoint that doesn't require authentication
    _url = '/api/v1.0/contacts/'
    
    # loop to activate the protection
    for i in range(0, 5):
        client.get(_url, HTTP_AUTHORIZATION=f"Bearer {jwt_token}")

    # this call should err
    response = client.get(_url, HTTP_AUTHORIZATION=f"Bearer {jwt_token}")
    # 429 - too many requests
    assert response.status_code == 429
```

Reply:
I made a very basic version of this test 
- instead of contacts, I use /users/ because I don't have to create anything
- I'm having a hard time overrind the throttle burst settings, despite using [related pytest doc](https://pytest-django.readthedocs.io/en/latest/configuring_django.html#overriding-individual-settings) and [django doc](https://docs.djangoproject.com/fr/4.2/topics/testing/tools/)

Reply:
That doesn't override ? 
```py
@override_settings(REST_FRAMEWORK={
  'DEFAULT_THROTTLE_RATES': {
    'burst': '2/min',
    'sustained': '2/min'
}})
```
The test passed on my side, so it must override it.

Reply:
Ok, indeed, keep what you have done. 
Like that it works, but it doesn't seem very clean:
```py
...
from rest_framework.decorators import api_view

@override_settings(REST_FRAMEWORK={
  'DEFAULT_THROTTLE_RATES': {
    'burst': '2/min',
    'sustained': '2/min'
}})
def test_throttle(request):
 ...
```

Reply:
On my side it only works when making 20+1 calls, the standard protection defined in settings.py. Any amount under 21 and the test fails (`assert 200 != 429`)

Reply:
Ok we investigated and, for the same code : 

- commands `make test` and `bin/pytest -k test_throttle` return this test failing, for assert 200 != 429
- command `make test-back core/tests/test_throttle.py` make it pass